sdk/identity/azure-identity/src/main/java/com/azure/identity/AuthFileCredential.java

```diff
@@ -0,0 +1,120 @@
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.core.util.serializer.JacksonAdapter;
+import com.azure.core.util.serializer.SerializerAdapter;
+import com.azure.core.util.serializer.SerializerEncoding;
+import com.azure.identity.implementation.IdentityClientOptions;
+import reactor.core.publisher.Mono;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Enables authentication to Azure Active Directory using configuration
+ * information stored Azure SDK Auth File.
+ *
+ * An Azaure SDK Auth file may be generated by passing <code>--sdk-auth</code>
+ * when generating an service principal with
+ * <code>az ad sp create-for-rbac</code>. At this time
+ * <see cref="AuthFileCredential"/> only supports SDK Auth Files which contain a
+ * client secret, client certificates are not supported at this time.
+ */
+@Immutable
+public class AuthFileCredential implements TokenCredential {
+    /* The file path value. */
+    private final String filepath;
+    private  IdentityClientOptions identityClientOptions;
+    private TokenCredential credential;
+    private static final SerializerAdapter SERIALIZER_ADAPTER = JacksonAdapter.createDefaultSerializerAdapter();
+
+    /**
+     *  Creates an instance of the SdkAuthFileCredential class based on information in given SDK Auth file.
+     *  If the file is not found or there are errors parsing it, <see cref="GetToken(TokenRequestContext)"/>
+     *  and <see cref="GetToken(TokenRequestContext)"/> will throw a <see cref="Exception"/>
+     *  with details on why the file could not be used.
+     *
+     * @param filepath The path to the SDK Auth file.
+     * @param identityClientOptions the options for configuring the identity client
+     */
+    AuthFileCredential(String filepath, IdentityClientOptions identityClientOptions) {
+        Objects.requireNonNull(filepath, "'filepath' cannot be null.");
+        Objects.requireNonNull(identityClientOptions, "'identityClientOptions' cannot be null.");
+        this.filepath = filepath;
+        this.identityClientOptions=identityClientOptions;
+    }
+
+    /**
+     * Obtains a token from the Azure Active Directory service, using the specified client detailed specified in the SDK Auth file.
+     * This method is called by Azure SDK clients. It isn't intended for use in application code.  
+     * If the SDK Auth file is missing or invalid, this method throws a <see cref="AuthenticationFailedException"/> exception.
+     * <param name="requestContext">The details of the authentication request</param>
+     * <returns>An <see cref="AccessToken"/> which can be used to authenticate service client calls.</returns>
+     */
+    @Override
+    public Mono<AccessToken> getToken(TokenRequestContext request) {
+        try {
+            ensureCredential();
```

Comment:
I appreciate the effort trying to get the credential built regardless of the validity of the auth file. 
In other languages they've been failing fast - throw an error while building the credential, which we are also moving towards in this PR https://github.com/Azure/azure-sdk-for-java/pull/7246/files#diff-56093106beda983b28894478869221c6R128 in case of an invalid certificate file. 
I would recommend moving this to the constructor.